app/src/main/java/movierating/services/MovieServiceImpl1.java

```diff
@@ -14,7 +15,10 @@ public MovieServiceImpl1(MovieRepository movieRepository) {
 
 
     @Override
-    public void addMovie(Movie movie) {
+    public void addMovie(Movie movie) throws MovieReleasedException {
+        if (movieRepository.getMovies().containsValue(movie))
```

Comment:
Chance of null pointer exception when movieRepository.getMovies() returns null.

Reply:
This is taken care.